### ISSUE TITLE
fuse: missing mountpoint parameter

### ISSUE THREAD

**Issue by karl0ss** [fangfufu/httpdirfs#62]
Running on RaspberryPi 4.

```
Linux PiServer 5.10.17-v8+ #1414 SMP PREEMPT Fri Apr 30 13:23:25 BST 2021 aarch64 GNU/Linux

fusermount version: 2.9.9

```

Pulled down the project, installed the Debian 10 requirements and built using 'make' in the project folder, no apparent issues -  

```
make
rm -f ./.depend
cc -O2 -Wall -Wextra -Wshadow -rdynamic -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DVERSION=\"1.2.0\" `pkg-config --cflags-only-I gumbo libcurl fuse uuid expat` -MM src/fuse_local.c src/link.c src/main.c src/util.c src/cache.c src/network.c src/sonic.c -MF ./.depend;
cc  -O2 -Wall -Wextra -Wshadow -rdynamic -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DVERSION=\"1.2.0\" `pkg-config --cflags-only-I gumbo libcurl fuse uuid expat` `pkg-config --libs-only-L gumbo libcurl fuse uuid expat` -c -o main.o src/main.c
src/main.c: In function ‘parse_arg_list’:
src/main.c:173:35: warning: comparison is always true due to limited range of data type [-Wtype-limits]
                     &long_index)) != -1) {
                                   ^~
cc  -O2 -Wall -Wextra -Wshadow -rdynamic -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DVERSION=\"1.2.0\" `pkg-config --cflags-only-I gumbo libcurl fuse uuid expat` `pkg-config --libs-only-L gumbo libcurl fuse uuid expat` -c -o network.o src/network.c
cc  -O2 -Wall -Wextra -Wshadow -rdynamic -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DVERSION=\"1.2.0\" `pkg-config --cflags-only-I gumbo libcurl fuse uuid expat` `pkg-config --libs-only-L gumbo libcurl fuse uuid expat` -c -o fuse_local.o src/fuse_local.c
cc  -O2 -Wall -Wextra -Wshadow -rdynamic -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DVERSION=\"1.2.0\" `pkg-config --cflags-only-I gumbo libcurl fuse uuid expat` `pkg-config --libs-only-L gumbo libcurl fuse uuid expat` -c -o link.o src/link.c
cc  -O2 -Wall -Wextra -Wshadow -rdynamic -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DVERSION=\"1.2.0\" `pkg-config --cflags-only-I gumbo libcurl fuse uuid expat` `pkg-config --libs-only-L gumbo libcurl fuse uuid expat` -c -o cache.o src/cache.c
cc  -O2 -Wall -Wextra -Wshadow -rdynamic -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DVERSION=\"1.2.0\" `pkg-config --cflags-only-I gumbo libcurl fuse uuid expat` `pkg-config --libs-only-L gumbo libcurl fuse uuid expat` -c -o util.o src/util.c
cc  -O2 -Wall -Wextra -Wshadow -rdynamic -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DVERSION=\"1.2.0\" `pkg-config --cflags-only-I gumbo libcurl fuse uuid expat` `pkg-config --libs-only-L gumbo libcurl fuse uuid expat` -c -o sonic.o src/sonic.c
cc  -O2 -Wall -Wextra -Wshadow -rdynamic -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -DVERSION=\"1.2.0\" `pkg-config --cflags-only-I gumbo libcurl fuse uuid expat` `pkg-config --libs-only-L gumbo libcurl fuse uuid expat` -o httpdirfs main.o network.o fuse_local.o link.o cache.o util.o sonic.o -pthread -lgumbo -lcurl -lfuse -lcrypto -luuid -lexpat

```

But when I try and mount (I have changed my HTTP folder URL for this ticket)

` ./httpdirfs -f --cache https://myHTTPFolder/ /mnt/HTTP`

I get 

```
see httpdirfs -h for usage
fuse: missing mountpoint parameter
```

Not sure what to do, any ideas would be really appreciated.

Thanks

**Comment by fangfufu** [comment]
What is the output of your ``uname -a``? Which distribution of Linux are you on? I don't have that error when I compile.

**Comment by karl0ss** [comment]
top line posted is the output of uname -a 

`Linux PiServer 5.10.17-v8+ #1414 SMP PREEMPT Fri Apr 30 13:23:25 BST 2021 aarch64 GNU/Linux
`

Running the latest version of raspbian from their website.

**Comment by karl0ss** [comment]
To be able to move on, I have also tried installing this on an Unbuntu 18.04 VPS I have and seem to be able to get it to "work"..

For some context, I am trying to use this to connect to my HTTP Media Folder on AllDebrid, Now, this seeeeeems to have worked, but reports all my videos as "invalid links"

```
 LinkTable 0x7f7b68003c60 for https://myfiles.alldebrid.com/7dbhzgmkp1/links
--------------------------------------------
0 H 0  https://myfiles.alldebrid.com/7dbhzgmkp1/links
1 I 0 Raya.and.the.Last.Dragon.2021.2160p.10bit.HDR.BluRay.8CH.x265.HEVC-PSA.mkv https://myfiles.alldebrid.com/7dbhzgmkp1/links/Raya.and.the.Last.Dragon.2021.2160p.10bit.HDR.BluRay.8CH.x265.HEVC-PSA.mkv
2 I 0 The.Chronicles.Of.Riddick.2004.1080p.BluRay.x264-[YTS.AM].mp4 https://myfiles.alldebrid.com/7dbhzgmkp1/links/The.Chronicles.Of.Riddick.2004.1080p.BluRay.x264-%5BYTS.AM%5D.mp4
3 I 0 Riddick.2013.1080p.BluRay.x264.YIFY.mp4 https://myfiles.alldebrid.com/7dbhzgmkp1/links/Riddick.2013.1080p.BluRay.x264.YIFY.mp4
--------------------------------------------
LinkTable_print(): Invalid link count: 3

```

So, this does feel like it is really close to working for my needs, but not quite, is this a limitation? or something I could get working maybe?

**Comment by fangfufu** [comment]
I am not 100% sure why all your links are marked as invalid. Here is the code that decode links:
https://github.com/fangfufu/httpdirfs/blob/master/src/link.c#L107

Basically your link has to start with alphanumeric character.

**Comment by fangfufu** [comment]
If you give me an account to whatever you are using, I might be able to fix it.

**Comment by karl0ss** [comment]
Is there a way for me to send you a PM and I will send you a link so you can have a look....this would be really useful..

Still didn't work out how to get it to work on my Raspberry Pi 4..

**Comment by fangfufu** [comment]
If you run the 64-bit beta version of Raspberry Pi OS or whatever it is called, it should work. It is basically vanilla Debian with Raspberry Pi foundation kernel. I think PiServer has a different version of one of the library, which leads to that error.

**Comment by karl0ss** [comment]
Ok cool, did you get my email yesterday with the link to what I was trying to access?

Thank for your work..

**Comment by fangfufu** [comment]
Yes, I got your email yesterday. I will likely have time to investigate over the weekend, or the next weekend. Sorry, this is going to take some time. I am a bit busy these days.

**Comment by karl0ss** [comment]
Yeah no problem, I will continue to play as well, thanks

**Comment by fangfufu** [comment]
You put your server behind Cloudflare, and the HTTP header response does not provide content length, so the files are marked as invalid.  If you use the server address that is behind Cloudflare, it works as intended. This is not my software's problem.

**Comment by mkurz** [comment]
Note for others, since it took me a while to figure out what's going on:
Just in case you are running on arm / aarch64 (e.g. Apple Silicon M1 / Asahi Linux) and run into:
`fuse: missing mountpoint parameter`
That is because the current httpdirfs release 1.2.3 is not compatible with arm / aarch64, see #103. A fix has been merged already: #104.
Here is a workaround for Arch Linux users, until a new release arrives: https://aur.archlinux.org/packages/httpdirfs#comment-880071
